components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java

```diff
@@ -217,6 +217,7 @@ protected void doStop() throws Exception {
     for (Repository repository : repositories.values()) {
       log.debug("Stopping repository: {}", repository);
       repository.stop();
+      store.update(repository.getConfiguration());
```

Comment:
Configuration is updated when stopped (whole repository manager, so NX is stopped, not given repository) to persist the current cacheTokens, a configuration bit that may change (proxy repo = user expires proxy caches, m2 group repo = member change).

This is fishy to me at least, but could not figure out better way to make new cacheToken get into the configuration, as in case of m2 group it is changing in doConfigure of m2 group repository itself...